### PR TITLE
MotionBlur + Instances

### PR DESCRIPTION
**Pull request by CraigFeldspar**
Fixes Instances and ThinInstances to work with MotionBlurPostProcess

Also changes the way velocity is written, removing the need for only positive values

### CODE REVIEW

File: src/Meshes/mesh.ts [hunk 1]
```diff
@@ -71,14 +71,17 @@ class _InstanceDataStorage {
     public batchCache = new _InstancesBatch();
     public instancesBufferSize = 32 * 16 * 4; // let's start with a maximum of 32 instances
     public instancesBuffer: Nullable<Buffer>;
+    public instancesPreviousBuffer: Nullable<Buffer>;
```
Comment (deltakosh) [inline src/Meshes/mesh.ts:74]: So in order to not go over the 128 properties per object (which triggers the slow path in v8, please add all these new properties in the _instanceDataStorage

**Review by deltakosh — APPROVED** [review]
Looks good so far!

File: tests/validation/config.json [hunk 2]
```diff
@@ -1,1220 +1,1276 @@
...
+            "title": "Refraction local cube map PBR",
+            "renderCount": 1,
+            "playgroundId": "#RJN64I#5",
+            "referenceImage": "refraction-local-pbr.png"
+        },
+        {
+            "title": "Instances + motion blur",
+            "renderCount": 10,
+            "playgroundId": "#YB006J#402",
+            "excludedEngines": ["webgl1"],
```
Comment (Popov72) [inline tests/validation/config.json:1223]: You should also exclude the **webgpu** engine of all the "instances + motion blur" tests as we are currently limited to only 8 vertex buffers.

**Review by Popov72 — COMMENTED** [review]
LGTM!

File: src/Meshes/thinInstanceMesh.ts [hunk 3]
```diff
@@ -95,24 +95,27 @@ declare module "./mesh" {
         /** @hidden */
         _thinInstanceUpdateBufferSize(kind: string, numInstances?: number): void;
 
+        /** @hidden */
+        _createMatrixBuffer(kind: string, buffer: Nullable<Float32Array>, staticBuffer: boolean): Buffer;
```
[the remark this answers is not in the document]
Comment (CraigFeldspar) [reply src/Meshes/thinInstanceMesh.ts]: aggressive linter :)

File: src/Meshes/thinInstanceMesh.ts [hunk 4]
```diff
@@ -95,24 +95,27 @@ declare module "./mesh" {
...
-Mesh.prototype.thinInstanceAdd = function(matrix: DeepImmutableObject<Matrix> | Array<DeepImmutableObject<Matrix>>, refresh: boolean = true): number {
+Mesh.prototype.thinInstanceAdd = function (matrix: DeepImmutableObject<Matrix> | Array<DeepImmutableObject<Matrix>>, refresh: boolean = true): number {
     this._thinInstanceUpdateBufferSize("matrix", Array.isArray(matrix) ? matrix.length : 1);
 
     const index = this._thinInstanceDataStorage.instancesCount;
 
     if (Array.isArray(matrix)) {
         for (let i = 0; i < matrix.length; ++i) {
-            this.thinInstanceSetMatrixAt(this._thinInstanceDataStorage.instancesCount++, matrix[i], (i === matrix.length - 1) && refresh);
+            this.thinInstanceSetMatrixAt(this._thinInstanceDataStorage.instancesCount++, matrix[i], i === matrix.length - 1 && refresh);
```
[the remark this answers is not in the document]
Comment (CraigFeldspar) [reply src/Meshes/thinInstanceMesh.ts]: aggressive linter :)

**Comment by julien-moreau** [comment]
LGTM for the motion blur part!

**Review by sebavan — APPROVED** [review]

**Pull request merged**